packages/web/src/components/AppConnectionContextMenu/index.jsx

```diff
@@ -18,15 +19,24 @@ function ContextMenu(props) {
     disableReconnection,
   } = props;
   const formatMessage = useFormatMessage();
+  const queryClient = useQueryClient();
+
   const createActionHandler = React.useCallback(
     (action) => {
-      return function clickHandler(event) {
+      return async function clickHandler(event) {
         onMenuItemClick(event, action);
+
+        if (action.type !== 'viewFlows') {
```

Comment:
I think it would be better to list allowed action types like:
` ['test', 'reconnect', 'delete'].includes(action.type)`

so if ever any new action is added, then it won't cause query invalidation by default.